src/nidmm/system_tests/test_system_nidmm.py

```diff
@@ -1,338 +1,352 @@
-    # Calling Configure Trigger function and asserting True if any error occurred while function call.
-    try:
-        session.configure_trigger(nidmm.TriggerSource.IMMEDIATE)
-    except nidmm.Error:
-        assert True
-
-
-def test_method_self_test(session):
-    # We should not get an assert if self_test passes
-    session.self_test()
-
-
-def test_method_get_dev_temp(session):
-    temperature = session.get_dev_temp('')
-    assert 20 <= temperature <= 50
-
-
-def test_method_reset_with_defaults(session):
-    assert session.reset_with_defaults() is None
-
-
-def test_method_get_self_cal_supported(session):
-    assert session.get_self_cal_supported() in [True, False]
-
-
-def test_method_read_status(session):
-    backlog, status = session.read_status()
-    assert isinstance(backlog, int)
-    assert backlog == 0
-
-
-def test_fetch_error_while_not_initiated(session):
-    try:
-        session.fetch()
-        assert False
-    except nidmm.Error as e:
-        assert e.code == -1074118641   # called fetch before calling Initiate or after calling Abort
-
-
-def test_multi_point_acquisition_with_measurement_absolute(session):
-    session.configure_multi_point(4, 2)
-    session.configure_measurement_absolute(nidmm.Function.DC_VOLTS, 0.02, 0.001)
-    measurements = session.read_multi_point(8)
-    assert len(measurements) == 8
-
-
-def test_disable(session):
-    session.configure_measurement_digits(nidmm.Function.DC_VOLTS, 10, 5.5)
-    with session.initiate():
-        time.sleep(0.1)
-        backlog, acquisition_state = session.read_status()
-        assert acquisition_state == nidmm.AcquisitionStatus.FINISHED_WITH_BACKLOG
-        session.disable()
-        backlog, acquisition_state = session.read_status()
-        assert acquisition_state == nidmm.AcquisitionStatus.NO_ACQUISITION_IN_PROGRESS
-
-
-def test_fetch_multiple(session):
-    session.configure_measurement_digits(nidmm.Function.DC_VOLTS, 10, 5.5)
-    session.configure_multi_point(sample_count=10, trigger_count=1)
-    with session.initiate():
-        measurements = session.fetch_multi_point(5)
-        assert len(measurements) == 5
-        measurements = session.fetch_multi_point(5)
-        backlog, acquisition_state = session.read_status()
-        assert acquisition_state == nidmm.AcquisitionStatus.FINISHED_WITH_NO_BACKLOG
-
-
-def test_get_auto_range_value(session):
-    with session.initiate():
-        session.fetch()
-        auto_range_value_property = session.auto_range_value
-        assert auto_range_value_property == 300   # simulated device auto_range_value to maximum 300
-
-
-def test_get_cal_date_time(session):
-    last_cal = session.get_cal_date_and_time(0)
-    assert last_cal.month == 3
-    assert last_cal.day == 1
-    assert last_cal.year == 1940
-    assert last_cal.hour == 0
-    assert last_cal.minute == 0   # cal_date_and_time should be 03/01/1940:00:00 for simulated 408x devices; 407x and 4065 returns 00/00/0000:00:00
+class SystemTests:
```

Comment:
it's unfortunate that these diffs are impossible to follow, we should have done the "turn into a class without functional differences" work in its own PR.

Reply:
It still doesn't help much unless you pull the branch locally and use a real diff tool. I split apart the first test change such that the system tests are first put into a class. You can see that commit in isolation [here](https://github.com/ni/nimi-python/pull/1833/commits/bb3808f2adc1a371465a5452088bb77b8ca959ce).

The next commit on top of that adds the gRPC testing, [here](https://github.com/ni/nimi-python/pull/1833/commits/35e5c4a8dc9813747a088c0ee30e2353e7ca8c4b#diff-916d5685a40780e03962ad6c406da272b854a8f97d5650a6892946f930fbcb86). Further changes to the tests can be seen [here](https://github.com/ni/nimi-python/pull/1833/commits/171928ba04deaa905b7dec77b42d0c5faf76688c), [here](https://github.com/ni/nimi-python/pull/1833/commits/d8995eba542f57891ee98573ecf4573547ae66e8), and [here](https://github.com/ni/nimi-python/pull/1833/commits/64b15b278ca4457ee6dd55689d9c3aa27dd6452d). All of these changes diff pretty well, at least.